project/src/store/selectors.ts

```diff
@@ -1,27 +1,36 @@
 import { filteredOffersByCity } from '../utils/utils';
-import { City, Offers } from '../types/types';
+import { City, OfferType } from '../types/types';
+import { createSelector } from '@reduxjs/toolkit';
 
 export const getFilteredOffers = (state: {
   currentCity: City;
-  offersList: Offers;
-}) => filteredOffersByCity(state.currentCity, state.offersList);
+  offers: { [offerId: number]: OfferType};
+}) => {
+  const filteredOffers = filteredOffersByCity(state.currentCity, state.offers);
+  return filteredOffers;
+};
 
 
-export const getSortedOffers = (state: {
-  currentSorting: string;
-  offersList: Offers;
-  currentCity: City;
-},
-) => {
-  const filteredOffers = filteredOffersByCity(state.currentCity, state.offersList);
-  switch (state.currentSorting) {
-    case 'Price: low to high':
-      return filteredOffers.sort((a, b) => a.price - b.price);
-    case 'Price: high to low':
-      return filteredOffers.sort((a, b) => b.price - a.price);
-    case 'Top rated first':
-      return filteredOffers.sort((a, b) => b.rating - a.rating);
-    default:
-      return filteredOffers;
+const getSortingName = (state: { currentSorting: string}) => state.currentSorting;
+
+export const getSortedOffers = createSelector(
+  getFilteredOffers,
+  getSortingName,
+  (filteredOffers, currentSorting) => {
+    switch (currentSorting) {
+      case 'Price: low to high':
+        return filteredOffers.sort((a, b) => a.price - b.price);
+      case 'Price: high to low':
+        return filteredOffers.sort((a, b) => b.price - a.price);
+      case 'Top rated first':
+        return filteredOffers.sort((a, b) => b.rating - a.rating);
+      default:
+        return filteredOffers;
+    }
   }
-};
+);
+
+
+export const getOfferById = (offerId:number) => (store:{
+  offers: { [offerId: number]: OfferType};
+}) => Object.values(store.offers).filter(({ id }) => id === offerId)[0];
```

Comment:
для поиска элемента в массиве лучше подходит [find](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find)

тогда и возвращаемый тип селектора будет правильный – `OfferType | undefined`

и ошибку в моём соседнем комментарии TS поможет найти
```suggestion
}) => Object.values(store.offers).find(({ id }) => id === offerId);
```